### PR TITLE
[release-4.12] Run csi-proxy on Windows nodes

### PR DESCRIPTION
**Pull request by sebsoto**
Includes backports for csi-proxy, as well as SMB CSI testing, in order to validate the functionality of csi-proxy.

### CODE REVIEW

File: .gitmodules
Comment (aravindhp) [inline .gitmodules:1]: Please fix the commit message

**Review by aravindhp — APPROVED** [review]

**Comment by aravindhp** [comment]
/hold

For merge ordering

**Comment by mansikulkarni96** [comment]
/retest-required
daemonset scale error

**Comment by mansikulkarni96** [comment]
/retest-required
`time="2023-09-17T16:55:15Z" level=fatal msg="Failed to run bundle: error waiting for CSV to install: rpc error: code = Unknown desc = malformed header: missing HTTP content-type\n"`

**Comment by mansikulkarni96** [comment]
Blocked by https://github.com/openshift/windows-machine-config-operator/pull/1811

**Pull request merged**